scheduler/scheduler-node/src/main/java/org/ow2/proactive/resourcemanager/task/client/RMNodeClient.java

```diff
@@ -0,0 +1,651 @@
+import org.ow2.proactive.resourcemanager.common.RMState;
+import org.ow2.proactive.resourcemanager.common.event.RMNodeHistory;
+import org.ow2.proactive.resourcemanager.common.event.RMNodeSourceEvent;
+import org.ow2.proactive.resourcemanager.common.event.dto.RMStateDelta;
+import org.ow2.proactive.resourcemanager.common.event.dto.RMStateFull;
+import org.ow2.proactive.resourcemanager.exception.RMActiveObjectCreationException;
+import org.ow2.proactive.resourcemanager.exception.RMException;
+import org.ow2.proactive.resourcemanager.exception.RMNodeException;
+import org.ow2.proactive.resourcemanager.frontend.topology.TopologyInfo;
+import org.ow2.proactive.resourcemanager.nodesource.common.NodeSourceConfiguration;
+import org.ow2.proactive.resourcemanager.nodesource.common.PluginDescriptor;
+import org.ow2.proactive.scheduler.common.exception.NotConnectedException;
+import org.ow2.proactive.scheduler.rest.IRMClient;
+import org.ow2.proactive.scripting.ScriptResult;
+import org.ow2.proactive_grid_cloud_portal.common.RMRestInterface;
+import org.ow2.proactive_grid_cloud_portal.common.dto.LoginForm;
+import org.ow2.proactive_grid_cloud_portal.rm.client.RMRestClient;
+import org.ow2.proactive_grid_cloud_portal.scheduler.exception.PermissionRestException;
+import org.ow2.proactive_grid_cloud_portal.scheduler.exception.RestException;
+
+
+@PublicAPI
+public class RMNodeClient implements IRMClient, Serializable {
+
+    private String schedulerRestUrl;
+
+    private CredData userCreds;
+
+    private RMRestInterface rm;
+
+    private ConnectionInfo connectionInfo;
+
+    private String sessionId;
+
+    public RMNodeClient(CredData userCreds, String schedulerRestUrl) {
+        this.schedulerRestUrl = schedulerRestUrl;
+        this.userCreds = userCreds;
+    }
+
+    public void connect() throws Exception {
+        init(new ConnectionInfo(schedulerRestUrl, userCreds.getLogin(), userCreds.getPassword(), null, true));
+    }
+
+    public void connect(String url) throws Exception {
+        schedulerRestUrl = url;
+        connect();
+    }
+
+    @Override
+    public void init(ConnectionInfo connectionInfo) throws Exception {
+        HttpClient client = new HttpClientBuilder().insecure(connectionInfo.isInsecure()).useSystemProperties().build();
+
+        RMRestClient restApiClient = new RMRestClient(connectionInfo.getUrl(), new ApacheHttpClient4Engine(client));
+
+        this.rm = restApiClient.getRm();
+
+        this.connectionInfo = connectionInfo;
+
+        renewSession();
+    }
+
+    public void renewSession() {
+        try {
+            String sid = rm.rmConnect(connectionInfo.getLogin(), connectionInfo.getPassword());
+            setSession(sid);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public ConnectionInfo getConnectionInfo() {
+        return connectionInfo;
+    }
+
+    @Override
+    public void setSession(String sid) {
```

Comment:
I think the setSession cannot be used like in the schedulerNodeClient, better remove this method

Reply:
ok

Reply:
because setSession does nothing. In the SchedulerNodeClient, one can do
schedulerapi.setSession("mycustomid")
schedulerapi.renewSession()
And then use the api.

Here it is not possible to do so

Reply:
here we can setSession. and then call anymethod, and new sessionid would be used

Reply:
Ok I didn't see it, but please note that this.rm is set in the init method only, so you need to call init first.

Reply:
actually no need to call init. we can just change session id and become another user of the same url.